dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -141,13 +141,9 @@ public async Task DeleteCollectionAsync(string collectionName, CancellationToken
     public async IAsyncEnumerable<MemoryRecord> GetBatchAsync(string collectionName, IEnumerable<string> keys, bool withEmbeddings = false,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        foreach (var key in keys)
+        await foreach (var result in this.InternalGetBatchAsync(collectionName, keys.ToArray(), withEmbeddings, cancellationToken).ConfigureAwait(false))
```

Comment:
No need to loop over the results here, you can just return this.InternalGetBatchAsync(collectionName, keys.ToArray(), withEmbeddings, cancellationToken), if you remove the async from the method signature.